integration/solana/setup.ts

```diff
@@ -6,30 +6,35 @@ import fs from 'fs';
 
 const endpoint: string = process.env.RPC_URL || "http://127.0.0.1:8899";
 
-export async function loadContract(name: string, args: any[] = [], space: number = 8192):
+export async function loadContractAndCallConstructor(name: string, args: any[] = [], space: number = 8192):
     Promise<{ program: Program, payer: Keypair, provider: AnchorProvider, storage: Keypair, program_key: PublicKey }> {
 
-    const idl = JSON.parse(fs.readFileSync(`${name}.json`, 'utf8'));
-
-    const payer = loadKey('payer.key');
+    const {program, payer, provider, program_key} = await loadContract(name);
 
     process.env['ANCHOR_WALLET'] = 'payer.key';
```

Comment:
`loadContract()` already does this

Reply:
Yes, you're right. That slipped through.